File: 8_kyu/Convert_number_to_reversed_array_of_digits.go
package kata

// Converts number n to reversed array of digits.
func Digitize(n int) []int {
	if n < 0 {
    // Also return the digits of negative numbers.
		n = -n
	}
	if n == 0 {
    // Special case 0
		return []int{0}
	}
  // Calculate the decimal digits of a positive number n.
	accu := make([]int, 0, 20)
	for n > 0 {
		accu = append(accu, n%10)
		n /= 10
	}
	return accu
}

File: 8_kyu/Convert_number_to_reversed_array_of_digits.r
#
# Convert number to reversed array of decimal digits.
#
digitize <- function(n) {
  stopifnot(is.numeric(n), n >= 0)
  v <- numeric(0)
  while (n > 0) {
    v <- append(v, n %% 10)
    n <- n %/% 10
  }
  if(length(v) > 0) v else c(0)
}